app/scripts/modules/core/src/search/widgets/Filters.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { Key } from '@spinnaker/core';
+
+import { Filter } from './Filter';
+import { IFilterType } from './SearchFilterTypeRegistry';
+
+import './filterlist.less';
+
+export interface IFiltersLayout {
+  header: string;
+  filterTypes: IFilterType[];
+}
+
+export interface IFiltersProps {
+  activeFilter: IFilterType;
+  layouts: IFiltersLayout[];
+  isOpen: boolean;
+  onFilterChange: (key: Key) => void;
+}
+
+@autoBindMethods
+export class Filters extends React.Component<IFiltersProps> {
+
+  constructor(props: IFiltersProps) {
+    super(props);
+  }
+
+  private handleKeyUp(event: React.KeyboardEvent<HTMLElement>): void {
+    switch (event.key) {
+      case Key.UP_ARROW:
+      case Key.DOWN_ARROW:
+        this.props.onFilterChange(event.key);
+        break;
+    }
+  }
+
+  private generateFilterElement(filterType: IFilterType): JSX.Element {
+
+    const { text, modifier } = filterType;
+    return (
+      <Filter
+        key={[text, modifier].join('|')}
+        filterType={filterType}
+        isActive={this.props.activeFilter.modifier === modifier}
+        onKeyUp={this.handleKeyUp}
+      />
+    );
+  }
+
+  private generateFilterLayout(layout: IFiltersLayout): JSX.Element {
```

Comment:
I think the term `render` is well enough understood in react-land as "return an element", since that is what the full `render()` function does.